controllers/module_nmc_reconciler.go

```diff
@@ -0,0 +1,150 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/filter"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/module"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/nmc"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+//+kubebuilder:rbac:groups="core",resources=nodes,verbs=get;watch;patch
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=nodemodulesconfigs,verbs=get;list;watch;update;patch;create
+
+const (
+	ModuleNMCReconcilerName = "ModuleNMCReconciler"
+)
+
+type ModuleNMCReconciler struct {
+	kernelAPI   module.KernelMapper
+	filter      *filter.Filter
+	reconHelper moduleNMCReconcilerHelperAPI
+	nmcHandler  nmc.ModuleNMCHandler
+}
+
+func NewModuleNMCReconciler(client client.Client,
+	kernelAPI module.KernelMapper,
+	nmcHandler nmc.ModuleNMCHandler,
+	filter *filter.Filter) *ModuleNMCReconciler {
+	reconHelper := newModuleNMCReconcilerHelper(client)
+	return &ModuleNMCReconciler{
+		kernelAPI:   kernelAPI,
+		filter:      filter,
+		reconHelper: reconHelper,
+		nmcHandler:  nmcHandler,
+	}
+}
+
+func (mnr *ModuleNMCReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// get the reconciled module
+	logger := log.FromContext(ctx)
+
+	logger.Info("Starting Module-NMS reconcilation", "module name and namespace", req.NamespacedName)
```

Comment:
Who says that the reconciled object is a `Module`? Can't it be a `Node`? a `Job`?

Reply:
because this is how this reconciler is defined. It is defined For Module, and for every "watch" it finds the appropriate Module to reconcile